x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialRelatesFunction.java

```diff
@@ -117,6 +133,93 @@ protected boolean geometryRelatesGeometry(GeometryDocValueReader reader, Compone
             return visitor.matches();
         }
 
+        protected void processSourceAndConstantGrid(
+            BooleanBlock.Builder builder,
+            int position,
+            BytesRefBlock wkb,
+            long gridId,
+            DataType gridType
+        ) {
+            if (wkb.getValueCount(position) < 1) {
+                builder.appendNull();
+            } else {
+                builder.appendBoolean(compareGeometryAndGrid(asGeometry(wkb, position), gridId, gridType));
+            }
+        }
+
+        protected void processSourceAndSourceGrid(
+            BooleanBlock.Builder builder,
+            int position,
+            BytesRefBlock wkb,
+            LongBlock gridIds,
+            DataType gridType
+        ) {
+            if (wkb.getValueCount(position) < 1 || gridIds.getValueCount(position) < 1) {
+                builder.appendNull();
+            } else {
+                builder.appendBoolean(compareGeometryAndGrid(asGeometry(wkb, position), gridIds.getLong(position), gridType));
+            }
+        }
+
+        protected boolean compareGeometryAndGrid(Geometry geometry, long gridId, DataType gridType) {
+            if (geometry instanceof Point point) {
+                long geoGridId = getGridId(point, gridId, gridType);
+                return gridId == geoGridId;
+            } else {
+                throw new IllegalArgumentException(
```

Comment:
This should be easy to implement though. It is ok to do it in a follow up

Reply:
More over, I think we should do it generic, no point in distinguish a point from another geometry, we have code that that this already.